### PR TITLE
Switch to GeoIP2, add db update and locale options.

### LINKED ISSUE #1745: Switching to GeoLite2
EssentialsX version (run `/essentials`): 2.0.1 build 562

Details:
The Legacy GeoLite dat will be discontinued by April 1, 2018. So it is the time to upgrade EssentialsGeoIP to work with GeoLite2 and MaxMind DB.

```
The free versions of the GeoLite Legacy databases available at 
http://dev.maxmind.com/geoip/legacy/geolite/ will no longer be updated 
and free downloads of GeoLite Legacy using the geoipupdate program 
will no longer be available starting April 1, 2018.
```
[https://support.maxmind.com/geolite-legacy-discontinuation-notice/](
https://support.maxmind.com/geolite-legacy-discontinuation-notice/)

### PR DESCRIPTION
**Pull request by k-jiang**
Fix #1745 #1746
 
Since [MaxMind is going to drop support on Legacy GeoIP db files](https://support.maxmind.com/geolite-legacy-discontinuation-notice/), I decided to upgrade EssentialsGeoIP to work with the newer and more featured GeoIP2 API.

The change of this PR include:
1. switch to GeoIP2 API
2. add option to update GeoIP2 db files
3. add option to show geolocation with locale

Some minor changes include:
1. update the locale files to reflect the correct size of new mmdb files.

Changes that I have thought of but did not included:
1. Custom message when query with GeoIP2 API generates exceptions.
GeoIP2 API now support several exception messages, including the "must have" `AddressNotFoundException` (see EssentialsGeoIPPlayerListener.jar line 97 to 106). It should be a good idea to add some custom messages when such a exception is caught. Yet I'm not sure if it is a good idea to do so since I could only handle English and Chinese (CN, HK and TW).

If you think there are option(s) I have added are unnecessary, please contact me and I will remove them removed for you.

### CODE REVIEW

**Comment by drtshock** [comment]
This brings `EssentialsXGeoIP.jar` up from 117.31kb to 4.3mb. Can we shade less of apache commons, or is there an alternative solution to untarring files?

File: EssentialsGeoIP/pom.xml [hunk 1]
```diff
@@ -26,14 +26,57 @@
 
     <dependencies>
         <dependency>
-            <groupId>net.ess3</groupId>
```
Comment (drtshock) [inline EssentialsGeoIP/pom.xml]: You removed essentials from the dependencies. Breaks this.

**Comment by k-jiang** [comment]
Well 1.6mb is the best I could do so far.
Now the remaining problem is, the official GeoIP2 API uses a dependency called `jackson-databind` which is bloated. This dependency itself occupies 1.2-1.3mb in size. I even tried to downgrade it as well just to gain some extra 200k advantage. Notice that v2.7.x is the lowest version it could run.

I think the other way out is to rewrite the GeoIP2 API by hand, but it is definitely going to be as pain as hell.

Let me know if you have any more idea.

File: EssentialsGeoIP/pom.xml (hunk 1 above)
Comment (k-jiang) [reply EssentialsGeoIP/pom.xml]: Some how,  with this dependency, maven includes the whole ess into GeoIP.jar. To be honest I'm not sure how to do it properly. At least it doesn't cause problem when I tested it :P

**Comment by k-jiang** [comment]
Ok I think I have a proper solution now. Trying to use `maven-shade-plugin`, which should get the job done.
But I need more time on testing, may push a new commit by tomorrow.

**Comment by drtshock** [comment]
Sweet. Thank you for your work on this ❤️

**Comment by k-jiang** [comment]
I think that's it. 1.1Mb is the best I could make. (~~F the jackson lib~~).

P.S. There is an interesting maven plugin called `com.github.wvengen:proguard-maven-plugin` which claimed to be able to shrink jar. Have tried and shrink the jar to ~700k but it generate exception during loading. Don't know exactly how to use it. But so far I think 1.1Mb is good enough. (~~F the jackson again~~).

**Comment by drtshock** [comment]
Nice. I'll take a look at this soon. Lets see if we can get @SupaHam to approve :)

**Review by DoNotSpamPls — APPROVED** [review]

**Comment by SupaHam** [comment]
Please remove javatar dependency and autotest.sh.

I'll need to test this later to make sure it works fine.

**Comment by k-jiang** [comment]
@SupaHam `javatar` is required to unpack maxmind's mmdb2 file from .tar.gz. You can check [here](https://dev.maxmind.com/geoip/geoip2/geolite2/#Downloads) for their new packing format.

I don't know why you want to remove it. Maybe you want me to use some other alternative instead?

**Comment by SupaHam** [comment]
Thank you very much for reaching out to us before the breakages happened.

The reason I wanted javatar removed is because of jar footprint, we need to make sure we're not adding things needlessly without considering better alternatives.

**Pull request merged**